File: 1657. Determine if Two Strings Are Close/closeStrings.cpp
class Solution
{
public:
    bool closeStrings(string word1, string word2)
    {

        vector<int> f1(26), f2(26), w1(26), w2(26);

        for (auto it : word1)
        {
            f1[it - 'a']++, w1[it - 'a'] = 1;
        }
        for (auto it : word2)
        {
            f2[it - 'a']++, w2[it - 'a'] = 1;
        }
        sort(f1.begin(), f1.end());
        sort(f2.begin(), f2.end());

        return f1 == f2 && w1 == w2;
    }
};

File: 1657. Determine if Two Strings Are Close/closeStrings.java
class Solution {
    public boolean closeStrings(String word1, String word2) {
        
        int f1[]  = new int[26];
        int f2[]  = new int[26];
        int w1[]  = new int[26];
        int w2[]  = new int[26];

        for(char c:word1.toCharArray()){
            f1[c-'a']++;
            w1[c-'a'] = 1;
        }
        for(char c: word2.toCharArray()){
            f2[c-'a']++;
            w2[c-'a'] = 1;
        }

        Arrays.sort(f1);
        Arrays.sort(f2);

        return Arrays.equals(f1,f2) && Arrays.equals(w1,w2);
    }
}